app/controllers/feeds_controller.rb

```diff
@@ -4,10 +4,16 @@ def new
   end
 
   def show
-    @feed = Feed.find_by!(token: params[:id])
+    @feed = Feed.find_by!(token: params[:id] || request.subdomain)
     @feed.touch(:fetched_at) if request.format == :atom
   end
 
+  def favicon
+    return redirect_to helpers.asset_path("favicon.ico") if request.subdomain.blank?
+
+    redirect_to Feed.find_by!(token: request.subdomain).favicon_url, allow_other_host: true
```

Comment:
How confused is 300ing the favicon going to make feed readers? Hopefully not at all, but I can see this needing more attention (proxying the favicon data? storing it in the db?) down the line 😬

Reply:
we can switch to proxying it as a one-line change, but storing it in our DB is probably the way to go in the end.